### ISSUE TITLE
Add support for kerberos auth over SSL

### ISSUE THREAD

**Issue by stonith** [WinRb/WinRM#86]
Currently this gem doesn't seem to support kerberos via SSL.

**Comment by danielladd** [comment]
I would also like to see kerberos over SSL supported.

**Comment by alv000h** [comment]
+1 @mwrock @sneal 
In winrm v1.8.1 seems to work.

I am using winrm v1.8.1 and I see that kerberos + SSL transport works with below endpoint configuration:
`endpoint: 'https://myhost:5986/wsman'`

The point is that kerberos + SSL Transport not support Self-signed Certificates because not initialize **"no_ssl_peer_verification!".**

Is there any problem if I open a Pull request to apply it in v1.8.1? I think it is compatible with 2.1.1 too but I cannot try it because I do not manage to connect using kerberos transport (similar problem as above).

I added this line to support it:
```WinRM]# git diff lib/winrm/http/transport.rb
diff --git a/lib/winrm/http/transport.rb b/lib/winrm/http/transport.rb
index 4ed9a0c..017522a 100644
--- a/lib/winrm/http/transport.rb
+++ b/lib/winrm/http/transport.rb
@@ -271,6 +271,7 @@ module WinRM
         no_sspi_auth!
         service ||= 'HTTP'
         @service = "#{service}/#{@endpoint.host}@#{realm}"
+        no_ssl_peer_verification! if opts[:no_ssl_peer_verification]
         init_krb
       end
```

This is the version (tag v1.8.1)
```
WinRM]# git log
commit 3b64b851d92c3d7d77a20c9bee7e351c168c7025
Merge: 34437a8 143afde
Author: Matt Wrock <matt@mattwrock.com>
Date:   Fri May 6 22:17:29 2016 -0700

    Merge pull request #211 from WinRb/1_8_1_rel
    Release 1.8.1
```

**Comment by juanjomm** [comment]
+1 @mwrock @sneal @alv000h
We are working in the same line and we are blocked in the same point.
It would be great to see kerberos over SSL supported.

**Comment by mwrock** [comment]
This should be fixed via #245
